packages/applications/ssr/src/components/organisms/garantiesFinancières/types.ts

```diff
@@ -0,0 +1,34 @@
+import { Iso8601DateTime } from '@potentiel-libraries/iso8601-datetime';
+
+export type GarantiesFinancièresActuelles = {
+  isActuelle: true;
+  type: string;
+  dateÉchéance?: Iso8601DateTime;
+  dateConstitution?: Iso8601DateTime;
+  attestation?: string;
+  validéLe?: Iso8601DateTime;
+  soumisLe?: Iso8601DateTime;
+  dernièreMiseÀJour: {
+    date: Iso8601DateTime;
+    par?: string;
+  };
+  actions: Array<
+    | 'enregister-attestation'
+    | 'demander-mainlevée-gf-pour-projet-abandonné'
+    | 'demander-mainlevée-gf-pour-projet-achevé'
+    | 'modifier'
+  >;
+};
+
+export type DépôtGarantiesFinancières = {
+  isActuelle: false;
+  type: string;
+  dateÉchéance?: Iso8601DateTime;
+  dateConstitution: Iso8601DateTime;
+  attestation: string;
+  dernièreMiseÀJour: {
+    date: Iso8601DateTime;
+    par?: string;
+  };
+  actions: Array<'instruire' | 'supprimer' | 'modifier'>;
+};
```

Comment:
J'aurais même poussé à peine plus loin en créant un type interne `CommonGarantiesFinancières` qui reprend les clés qu'on retrouve pour le dépot + les actuelles.


```suggestion
type CommonGarantiesFinancières = {
  type: string;
  dateÉchéance?: Iso8601DateTime;
  dateConstitution: Iso8601DateTime;
  dernièreMiseÀJour: {
    date: Iso8601DateTime;
    par?: string;
  };
}


export type GarantiesFinancièresActuelles = CommonGarantiesFinancières & {
  isActuelle: true;
  attestation?: string;
  validéLe?: Iso8601DateTime;
  soumisLe?: Iso8601DateTime;
  actions: Array<
    | 'enregister-attestation'
    | 'demander-mainlevée-gf-pour-projet-abandonné'
    | 'demander-mainlevée-gf-pour-projet-achevé'
    | 'modifier'
  >;
};

export type DépôtGarantiesFinancières = CommonGarantiesFinancières & {
  isActuelle: false;
  attestation: string;
  actions: Array<'instruire' | 'supprimer' | 'modifier'>;
};
```

Reply:
Oui pas bête, je regarde pour améliorer ça